test/ServiceRegistry.js

```diff
@@ -1147,6 +1159,263 @@ describe("ServiceRegistry", function () {
+                        { type: "uint256", name: "safeTxGas" },
+                        { type: "uint256", name: "baseGas" },
+                        { type: "uint256", name: "gasPrice" },
+                        { type: "address", name: "gasToken" },
+                        { type: "address", name: "refundReceiver" },
+                        { type: "uint256", name: "nonce" },
+                    ]
+                };
+
+                // Get the signature of a multisend transaction
+                const signatureBytes = await serviceOwner._signTypedData({verifyingContract: multisig.address, chainId: chainId}, EIP712_SAFE_TX_TYPE, safeTx);
+
+                // Forming Gnosis Safe transaction bytes data: the multisig address itself plus the execTransaction() data
+                // that forms the multisend data with the signature bytes to be executed by the multisig proxy
+                const safeExecData = gnosisSafe.interface.encodeFunctionData("execTransaction", [safeTx.to, safeTx.value,
+                    safeTx.data, safeTx.operation, safeTx.safeTxGas, safeTx.baseGas, safeTx.gasPrice, safeTx.gasToken,
+                    safeTx.refundReceiver, signatureBytes]);
+
+                // Add the multisig address on top of the multisig exec transaction data
+                const packedData = ethers.utils.solidityPack(["address", "bytes"], [multisig.address, safeExecData]);
+
+                // Try to deploy when passing the incorrect payload of a multisend
+                let safeExecErrData = gnosisSafe.interface.encodeFunctionData("execTransaction", [safeTx.to, safeTx.value,
+                    "0xabcd", safeTx.operation, safeTx.safeTxGas, safeTx.baseGas, safeTx.gasPrice, safeTx.gasToken,
+                    safeTx.refundReceiver, signatureBytes]);
+                let packedErrData = ethers.utils.solidityPack(["address", "bytes"], [multisig.address, safeExecErrData]);
+                await expect(
+                    serviceRegistry.connect(serviceManager).deploy(serviceOwnerAddress, serviceId,
+                        gnosisSafeSameAddressMultisig.address, packedErrData)
+                ).to.be.revertedWith("MultisigExecFailed");
+
+                // Try to deploy when passing the incorrect tx signature
+                safeExecErrData = gnosisSafe.interface.encodeFunctionData("execTransaction", [safeTx.to, safeTx.value,
+                    safeTx.data, safeTx.operation, safeTx.safeTxGas, safeTx.baseGas, safeTx.gasPrice, safeTx.gasToken,
+                    safeTx.refundReceiver, "0xabcd"]);
+                packedErrData = ethers.utils.solidityPack(["address", "bytes"], [multisig.address, safeExecErrData]);
+                await expect(
+                    serviceRegistry.connect(serviceManager).deploy(serviceOwnerAddress, serviceId,
+                        gnosisSafeSameAddressMultisig.address, packedErrData)
+                ).to.be.revertedWith("MultisigExecFailed");
```

Comment:
This two cases show that the safe transaction reverts with incorrectly constructed data fields.